extensions/sendbird/v1/actions/updateCustomerCustomFields/updateCustomerCustomFields.ts

```diff
@@ -40,10 +44,92 @@ export const updateCustomerCustomFields: Action<
         deskApiToken,
       })
 
-      await client.deskApi.updateCustomerCustomFields(customerId, {
+      const {
+        data: { customFields: receivedCustomFields },
+      } = await client.deskApi.updateCustomerCustomFields(customerId, {
         customFields,
       })
 
+      // check that all fields were updated
+      try {
+        const customFieldsObject = JSON.parse(customFields)
+        const providedFields = Object.keys(customFieldsObject)
+        const receivedFields = receivedCustomFields.map(({ key }) => key)
+        const missingFields = receivedFields.filter(
+          (field) => !providedFields.includes(field)
+        )
+        if (missingFields.length > 0) {
+          await onError({
+            events: [
+              {
+                date: new Date().toISOString(),
+                text: {
+                  en: `Some custom fields do not appear to have been set correctly`,
+                },
+                error: {
+                  category: 'WRONG_DATA',
+                  message: `The following fields were not updated: ${missingFields.join(
+                    ', '
+                  )}`,
+                },
+              },
+            ],
+          })
+          return
+        }
+      } catch (e) {
+        // if we can't parse the JSON, we can't check if all fields were updated
+        // so we just skip this check
+      }
+
+      // check that values match
+      try {
+        const customFieldsObject = JSON.parse(customFields)
+        const fieldsWithMismatchedValues = Object.keys(customFieldsObject)
+          .filter((key) => {
+            const valueToSet =
+              customFieldsObject[key as keyof typeof customFields]
```

Comment:
is this typecast necessary? shouldn't it just be a string given from Object.keys?

Reply:
keys is just an array of strings so to typescript this is just unchecked object access (i.e. you could be trying to access a random field that doesn't exist on the object since the string value is not types). casting it this was is saying, "hey ts, i know that these strings are the key names from the object, don't worry" 

if there's a better way, let me know, been looking for one for a while as casting is obviously a last resort